File: tindahan_natin/lib/features/products/product.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'product.freezed.dart';
part 'product.g.dart';

@freezed
abstract class Product with _$Product {
  const factory Product({
    required String id,
    required String name,
    required double price,
    required int quantity,
    required String categoryId,
    String? description,
    String? imageUrl,
    String? barcode,
    required String storeId,
    String? shelfId,
  }) = _Product;

  factory Product.fromJson(Map<String, dynamic> json) => _$ProductFromJson(json);
}

File: TindahanNatin.Server/Models/Product.cs
using System;
using NpgsqlTypes;

namespace TindahanNatin.Server.Models;

public class Product
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int Quantity { get; set; }
    public Guid CategoryId { get; set; }
    public Guid? ShelfId { get; set; }
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public string? Barcode { get; set; }
    public Guid StoreId { get; set; }

    // PostgreSQL full-text search vector (generated column)
    public NpgsqlTsVector? SearchVector { get; set; }
}